core/src/test/java/io/grpc/internal/RetriableStreamTest.java

```diff
@@ -280,14 +280,12 @@ public Void answer(InvocationOnMock in) {
     retriableStream.request(456);
 
     inOrder.verify(mockStream1).writeMessage(any(InputStream.class));
-    inOrder.verify(mockStream1).flush(); // Memory leak workaround
     inOrder.verify(mockStream1).writeMessage(any(InputStream.class));
-    inOrder.verify(mockStream1).flush(); // Memory leak workaround
     inOrder.verify(mockStream1).request(345);
     inOrder.verify(mockStream1, times(2)).flush();
     inOrder.verify(mockStream1).writeMessage(any(InputStream.class));
-    inOrder.verify(mockStream1).flush(); // Memory leak workaround
     inOrder.verify(mockStream1).request(456);
+    verifyNoMoreInteractions(mockStream1);
```

Comment:
I'm confused what is happening with the in-order invocation verification in this test. As written, the test fails and wants 3 `writeMessage()` invocations. Adding `times(3)` causes the test to fail with a missing `request()` invocation. Removing the two leading `writeMessage()` gets to the `verifyNoMoreInteractions()` and fails, printing:

```
For your reference, here is the list of all invocations ([?] - means unverified).
    1. -> at io.grpc.internal.RetriableStream$1AuthorityEntry.runWith(RetriableStream.java:685)
    2. -> at io.grpc.internal.RetriableStream$1CompressorEntry.runWith(RetriableStream.java:637)
    3. -> at io.grpc.internal.RetriableStream$1DecompressorRegistryEntry.runWith(RetriableStream.java:697)
    4. -> at io.grpc.internal.RetriableStream$1FullStreamDecompressionEntry.runWith(RetriableStream.java:649)
    5. -> at io.grpc.internal.RetriableStream$1FullStreamDecompressionEntry.runWith(RetriableStream.java:649)
    6. -> at io.grpc.internal.RetriableStream$1MaxInboundMessageSizeEntry.runWith(RetriableStream.java:709)
    7. -> at io.grpc.internal.RetriableStream$1MessageCompressionEntry.runWith(RetriableStream.java:661)
    8. -> at io.grpc.internal.RetriableStream$1MaxOutboundMessageSizeEntry.runWith(RetriableStream.java:721)
    9. -> at io.grpc.internal.RetriableStream$1MessageCompressionEntry.runWith(RetriableStream.java:661)
    10. -> at io.grpc.internal.RetriableStream$StartEntry.runWith(RetriableStream.java:361)
    11. -> at io.grpc.internal.RetriableStream.isReady(RetriableStream.java:613)
    12. [?]-> at io.grpc.internal.RetriableStream$1SendMessageEntry.runWith(RetriableStream.java:567)
    13. [?]-> at io.grpc.internal.RetriableStream$1SendMessageEntry.runWith(RetriableStream.java:567)
    14. -> at io.grpc.internal.RetriableStream$1RequestEntry.runWith(RetriableStream.java:585)
    15. -> at io.grpc.internal.RetriableStream$1FlushEntry.runWith(RetriableStream.java:603)
    16. -> at io.grpc.internal.RetriableStream$1FlushEntry.runWith(RetriableStream.java:603)
    17. -> at io.grpc.internal.RetriableStream$1SendMessageEntry.runWith(RetriableStream.java:567)
    18. -> at io.grpc.internal.RetriableStream$1RequestEntry.runWith(RetriableStream.java:585)
        at app//io.grpc.internal.RetriableStreamTest.retry_everythingDrained(RetriableStreamTest.java:288)
```

which indicates the two leading `writeMessage()` calls. What am I missing here?